test/Spec/MockContract.hs

```diff
@@ -268,6 +249,24 @@ mockQueryUtxoOut utxos =
             utxos
       ]
 
+valueToUtxoOut :: Value.Value -> Text
+valueToUtxoOut =
+  Text.intercalate " + " . map stringifyValue' . Value.flattenValue
+  where
+    stringifyValue' (curSymbol, tokenName, tAmt) =
+      let token =
+            if curSymbol == Ada.adaSymbol
+              then "lovelace"
+              else
+                let curSymbol' =
+                      encodeByteString $
+                        fromBuiltin $ Value.unCurrencySymbol curSymbol
+                    tokenName' =
+                      decodeUtf8 $
+                        fromBuiltin $ Value.unTokenName tokenName
+                 in [text|${curSymbol'}.${tokenName'}|]
```

Comment:
If the token name is empty, the dot shouldn't be there

Reply:
Aaa, true, I absolutely forgot! Thanks

Reply:
I'll write tests for this

Reply:
Nice, thanks :)

Reply:
I realised that I haven't backported your changes from the bulk airdrop repo. I fixed that. Also, I had to fix the UtxoParser too, so I ended up writing some tests for the parser.